src/test/java/net/sf/jabref/importer/fileformat/BibtexParserTest.java

```diff
@@ -1486,4 +1487,43 @@ public void parseReturnsEntriesInSameOrder() throws IOException {
 
         assertEquals(expected, result.getDatabase().getEntries());
     }
+
+    @Test
+    public void parsePrecedingComment() throws IOException {
+        // @formatter:off
+        String bibtexEntry = "% Some random comment that should stay here" + Globals.NEWLINE +
+                "@Article{test," + Globals.NEWLINE +
+                "  Author                   = {Foo Bar}," + Globals.NEWLINE +
+                "  Journal                  = {International Journal of Something}," + Globals.NEWLINE +
+                "  Note                     = {some note}," + Globals.NEWLINE +
+                "  Number                   = {1}" + Globals.NEWLINE +
+                "}";
+        // @formatter:on
+
+        // read in bibtex string
+        ParserResult result = BibtexParser.parse(new StringReader(bibtexEntry));
+
+        Collection<BibEntry> entries = result.getDatabase().getEntries();
+        assertEquals(1, entries.size());
```

Comment:
The `withField` syntax is something new for me.

Anyway, this will not work here, unfortunately. The crucial part is the comment before the entry and this cannot be set as a field.